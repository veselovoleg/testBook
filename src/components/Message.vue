<template>
  <li 
  class="message" 
  style="margin-top: 10px"
  @click="handleClick"
  >{{ message }}</li>
</template>

<script>
export default {
  props: {
    message: String,
    author: {
      type: String,
      default: 'Paco'
    }
  },
  methods: {
    handleClick () {
      this.$emit('message-clicked', this.message)
    }
  }
}
</script>