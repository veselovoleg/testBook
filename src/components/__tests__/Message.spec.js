import { mount } from "@vue/test-utils";
import Message from "@/components/Message.vue";

describe("Message.test.js", () => {
    let wrapper;
    const createWrapper = propsData => mount(Message, { propsData });

    describe("Properties", () => {
        it("has a message property", () => {
            wrapper = createWrapper({ message: "hey" });
            expect(wrapper.props("message", "hey")).toBeTruthy();
        });

        it("has no cat property", () => {
            wrapper = createWrapper({ cat: "hey" });
            expect(wrapper.props("cat", "hey")).toBeFalsy();
        });

        it("Paco is the default author", () => {
            wrapper = createWrapper({ message: "hey" });
            expect(wrapper.props("author", "Paco")).toBeTruthy();
        });
    });

    describe("Events", () => {
        beforeEach(() => {
            wrapper = createWrapper({ message: "Cat" });
        });

        it("has the expected html structure", () => {
            expect(wrapper.element).toMatchSnapshot();
        });

        it("calls handleClick", () => {
            const spy = jest.spyOn(wrapper.vm, 'handleClick');
            wrapper.vm.handleClick();
            expect(spy).toHaveBeenCalled();
        });

        //NOT WORKING. WHY?
        /*
        it("calls handleClick when click on message", () => {
            wrapper.vm.handleClick = jest.fn();
            expect(wrapper.contains('.message')).toBe(true);

            wrapper.find('.message').trigger('click');
            expect(wrapper.vm.handleClick).toHaveBeenCalledTimes(1);
        })
        */
    });
});