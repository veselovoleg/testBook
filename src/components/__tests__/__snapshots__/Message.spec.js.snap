// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message.test.js Events has the expected html structure 1`] = `
<li
  class="message"
  style="margin-top: 10px;"
>
  
  Cat

</li>
`;
